src/helpers/check-merge-safety.ts

```diff
@@ -61,6 +75,18 @@ const handlePushWorkflow = async (inputs: CheckMergeSafety) => {
   await map(filteredPullRequests, pullRequest => setMergeSafetyStatus(pullRequest as PullRequest, inputs));
 };
 
+const checkForExistingFailureStatus = async (pullRequest: PullRequest, context: string) => {
+  const { data } = await octokit.repos.getCombinedStatusForRef({
+    ...githubContext.repo,
+    ref: pullRequest.head.sha
+  });
+  if (data.state === 'failure') {
+    const existingContext = data.statuses.find(status => status.context === context);
+    return Boolean(existingContext);
+  }
+  return false;
```

Comment:
I see the micro-optimization here, but this seems simpler. Am I missing something?
```suggestion
  return Boolean(data.statuses.find(status => status.context === context));
```

Reply:
statuses will include `Merge Safety` even if the state is `success`, so we need to ensure we're looking at a `failure` before deciding not to re-set the status to `failure`. for a `failure`, failed statuses are listed, and for a `success`, all manually-set successful statuses are listed (afaik)

Reply:
ah ok, makes sense!